pandas/core/generic.py

```diff
@@ -4418,6 +4422,95 @@ def sort_values(
         """
         raise AbstractMethodError(self)
 
+    def _get_indexer(
+        self, target, level, ascending, kind, na_position, sort_remaining, key
+    ):
+        target = ensure_key_mapped(target, key, levels=level)
+        target = target._sort_levels_monotonic()
+
+        if level is not None:
+            _, indexer = target.sortlevel(
+                level, ascending=ascending, sort_remaining=sort_remaining
+            )
+        elif isinstance(target, MultiIndex):
+            from pandas.core.sorting import lexsort_indexer
+
+            indexer = lexsort_indexer(
+                target._get_codes_for_sorting(),
+                orders=ascending,
+                na_position=na_position,
+            )
+        else:
+            from pandas.core.sorting import nargsort
+
+            # Check monotonic-ness before sort an index (GH 11080)
```

Comment:
rigth can you move this to pandas.core.sorting (and name it like get_indexer_indexer)